### PR TITLE
feat: add support for distribute permission for release_bundle target.

### PR DESCRIPTION
**Pull request by martinm82**

### CODE REVIEW

**Comment by martinm82** [comment]
I have read the CLA Document and I hereby sign the CLA
#recheck

**Comment by martinm82** [comment]
recheck

**Review by alexhung — DISMISSED** [review]
Code looks good to me.

There are couple of PRs in review stage so we need to figure out the order of release.

**Comment by alexhung** [comment]
@chb0github ☝🏼

File: pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go [hunk 1]
```diff
@@ -121,7 +121,7 @@ const permissionFull = `
 		actions {
 			users {
 				name        = "anonymous"
-				permissions = ["read", "write"]
+				permissions = ["read", "write", "distribute"]
```
Comment (chb0github) [inline pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go:124]: We have 2 new permissions added and Only 1 added here

**Review by chb0github — CHANGES_REQUESTED** [review]
xray permission change must be tested.

Plus, [the docs I am looking at](https://www.jfrog.com/confluence/display/JFROG/Security+Configuration+JSON#SecurityConfigurationJSON-application/vnd.org.jfrog.artifactory.security.PermissionTarget+json) aren't clear. What documentation are you basing this on

**Comment by martinm82** [comment]
> xray permission change must be tested.
> 
> Plus, [the docs I am looking at](https://www.jfrog.com/confluence/display/JFROG/Security+Configuration+JSON#SecurityConfigurationJSON-application/vnd.org.jfrog.artifactory.security.PermissionTarget+json) aren't clear. What documentation are you basing this on

I mainly took the information out of the UI (permission target -> users/group -> ?)

> **Read**: Views and downloads Release Bundle artifacts from the relevant Release Bundle repository and reads the corresponding Release Bundles in the Distribution page 
> **Annotate**: Annotates Release Bundle artifacts and folder with metadata and properties 
> **Create**: Creates Release Bundles Distribute: Distributes release bundles 
> **Delete**: Deletes Release Bundles 
> **Manage Xray Metadata**: Triggers Xray scans on Release Bundles. Creates and deletes custom issues and license 
> **Manage**: Allows changing Release Bundle permission settings for other users in this permission target. It does not permit adding/removing resources to the permission target

File: pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go (hunk 1 above)
Comment (martinm82) [reply pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go:124]: I am not sure whether it makes actually sense to add here such permissions for the anonymous user.

**Comment by alexhung** [comment]
The V2 schema have all the permissions listed: https://www.jfrog.com/confluence/display/JFROG/Security+Configuration+JSON#SecurityConfigurationJSON-application/vnd.org.jfrog.artifactory.security.PermissionTargetV2+json

File: pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go (hunk 1 above)
Comment (alexhung) [reply pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go:124]: Let's create a new test (non-anon) user in that block of HCL and reference it in the permission target resource. Then we can include all the permissions and check for them, which currently we don't have any.

**Comment by martinm82** [comment]
Sorry for push forcing to this branch. I mixed up my GitHub accounts and therefore the CLA job failed.

**Review by chb0github — APPROVED** [review]
Looks good to me

**Comment by martinmigasiewicz-tomtom** [comment]
Do you have any ci running the acceptance tests?

**Comment by alexhung** [comment]
@martinm82 I found the issues.

First the test values must be string:

```
# github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/security_test [github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/security.test]
pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go:258:89: cannot use 3 (untyped int constant) as string value in argument to resource.TestCheckResourceAttr
pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go:264:90: cannot use 5 (untyped int constant) as string value in argument to resource.TestCheckResourceAttr
pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go:272:99: cannot use 4 (untyped int constant) as string value in argument to resource.TestCheckResourceAttr
FAIL	github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/security [build failed]
```

After fixing those, the HCL for the test has incorrect references. Where:

`name        = artifactory_managed_user.test-user`

Should be:

`name        = artifactory_managed_user.test-user.name`

for `permissionFull`.

Then, the user password does not meet requirement.

Finally the `TestAccPermissionTarget_user_permissions` test is missing a check for the `write` permission.

Do you want me to send you the patch file?

**Comment by martinm82** [comment]
> @martinm82 Here's the patch file: [0001-Fix-errors-in-tests.patch.zip](https://github.com/jfrog/terraform-provider-artifactory/files/8863420/0001-Fix-errors-in-tests.patch.zip)

Thanks @alexhung I will apply it tonight.

File: pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go (hunk 1 above)
Comment (chb0github) [reply pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go]: Did this get done?

**Comment by chb0github** [comment]
@alexhung @martinm82 - Where are we at on this gents? We gonna finish or close?

**Comment by martinmigasiewicz-tomtom** [comment]
Sorry gents, I haven't much time recently. I will get back to this tomorrow.

**Comment by martinm82** [comment]
@alexhung @chb0github PR is updated. Hopefully the tests will pass now

**Comment by alexhung** [comment]
@martinm82 One test failed. Not sure if I missed it last time or something else:

> # github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/security_test [github.com/jfrog/terraform-provider-artifactory/v6/pkg/artifactory/resource/security.test]
pkg/artifactory/resource/security/resource_artifactory_permission_target_test.go:254:21: undefined: acctest.ExecuteTemplate

At this point, my suggestion would be to merge this PR and then either me or @chb0github will fix this test with a separate PR.

**Comment by chb0github** [comment]
@alexhung that's been moved to the latest version of the shared package.

**Comment by chb0github** [comment]
@martinm82 if you change the package to `util` and import the shared repo (1.2.0) that should work

**Pull request merged**